solr/core/src/java/org/apache/solr/security/RuleBasedAuthorizationPluginBase.java

```diff
@@ -173,6 +173,7 @@ private boolean predefinedPermissionAppliesToRequest(Permission predefinedPermis
       log.trace("'ALL' perm applies to all requests; perm applies.");
       return true; //'ALL' applies to everything!
     } else if (! (context.getHandler() instanceof PermissionNameProvider)) {
+      // TODO: Is this code path needed anymore, now that all handlers implement the interface? context.getHandler returns Object and is not documented
```

Comment:
I though of removing this code, but did not dare to, as I'm not sure if `getHandler()` may perhaps still return a class  not implementing the interface...

Reply:
I'm fine leaving this in for the sake of caution.

That said, if we wanted to try removal, I guess we could rely on our tests a bit here - add temporary `throw new RuntimeException("blah")` here and see if any tests trigger it in local runs.

But maybe its not worth the hassle.